src/system.cpp

```diff
@@ -769,7 +769,10 @@ int System::CastOffPages(FunctorParams *functorParams)
         currentShift += params->m_pgHead2Height + params->m_pgFoot2Height;
     }
 
-    if ((params->m_currentPage->GetChildCount() > 0) && (this->m_drawingYRel - this->GetHeight() - currentShift < 0)) {
+    const auto systemMaxPerPage = params->m_doc->GetOptions()->m_systemMaxPerPage.GetValue();
+    const auto childsCount = params->m_currentPage->GetChildCount();
```

Comment:
We don't use `auto` in such cases. Rename `childsCount` into `childCount`

Reply:
Done.